website/src/app/kb/administer/upgrading/readme.mdx

```diff
@@ -73,13 +51,52 @@ necessary to upgrade the Gateway on service start. Simply move the old binary
 and restart the service to trigger the upgrade:
 
 ```
+# Stop the Gateway service. WARNING: This will cause brief downtime for this Gateway.
+sudo systemctl stop firezone-gateway
+
+# Move the old binary to a backup location
 sudo mv /usr/local/bin/firezone-gateway /usr/local/bin/firezone-gateway.bak
-sudo systemctl restart firezone-gateway
+
+# Start the Gateway service to trigger the upgrade
+sudo systemctl start firezone-gateway
 ```
 
+### Rollback in case of issues
+
+If something goes wrong during the upgrade, you can always revert to the old
+binary by stopping the service and moving the backup binary back to its original
+location:
+
+```
+# Restore the old binary and restart the service
+
+sudo systemctl stop firezone-gateway
+
+sudo mv /usr/local/bin/firezone-gateway.bak /usr/local/bin/firezone-gateway
+
+sudo systemctl start firezone-gateway
+```
+
+Please
+[file a GitHub issue](https://www.github.com/firezone/firezone/issues/new/choose)
+if you consistently encounter issues with the upgrade process.
+
 </TabsItem>
 </TabsGroup>
 
+### Verification
+
+After running the upgrade, you can verify the Gateway is r:wqaunning the latest
+version by checking what version it is reporting in the admin portal.
```

Comment:
What do you think about making this a little more concise?

```suggestion
After upgrading, verify the Gateway's version in the admin portal to ensure it's up to date.
```

Reply:
Updated